src/multivariate/mvnormal.jl

```diff
@@ -122,19 +117,19 @@ logdetcov(d::MvNormal) = logdet(d.Σ)
 
 ### Evaluation
 
-sqmahal(d::MvNormal, x::AbstractVector) = invquad(d.Σ, x - d.μ)
+sqmahal{T<:Real}(d::MvNormal, x::AbstractVector{T}) = invquad(d.Σ, x - d.μ)
 
-sqmahal!(r::AbstractVector, d::MvNormal, x::AbstractMatrix) =
+sqmahal!{T<:Real}(r::AbstractVector, d::MvNormal, x::AbstractMatrix{T}) =
     invquad!(r, d.Σ, x .- d.μ)
 
-gradlogpdf(d::MvNormal, x::Vector) = -(d.Σ \ (x - d.μ))
+gradlogpdf(d::MvNormal, x::Vector{Float64}) = -(d.Σ \ (x - d.μ))
 
 # Sampling (for GenericMvNormal)
 
-_rand!(d::MvNormal, x::VecOrMat) = add!(unwhiten!(d.Σ, randn!(x)), d.μ)
+_rand!(d::MvNormal, x::VecOrMat{Float64}) = add!(unwhiten!(d.Σ, randn!(x)), d.μ)
```

Comment:
Same. Better to let it fail in `randn!` if element type is not supported. You'll never know when somebody extends with a new method.